Source/Orts.Simulation/Simulation/RollingStocks/MSTSWagon.cs

```diff
@@ -1084,16 +1005,134 @@ public virtual void LoadFromWagFile(string wagFilePath)
+            {
+                // 1) First get the values from the zero load stage / base stage
+                if (mode != BrakeSystem.BrakeMode && BrakeSystems.TryGetValue((mode, 0), out var brakeSystem))
+                {
+                    HandleIncompatibleBrakesystems(brakeSystem);
+                    BrakeSystem.InitializeFromCopy(brakeSystem, true);
+                }
+
+                // 2) Then try to auto-switch to / get values from the appropriate load stage
+                var max = 0f;
+                foreach (var key in BrakeSystems.Keys)
+                    if (key.BrakeMode == mode && key.MinMass <= massKg)
+                        max = Math.Max(max, key.MinMass);
+                if (BrakeSystems.TryGetValue((mode, max), out brakeSystem))
+                {
+                    HandleIncompatibleBrakesystems(brakeSystem);
+                    BrakeSystem.InitializeFromCopy(brakeSystem, true);
+
+                    LoadEmptyMaxHandbrakeForceN = brakeSystem.InitialMaxHandbrakeForceN;
+                    LoadEmptyMaxBrakeForceN = BrakeShoeType != BrakeShoeTypes.Unknown && brakeSystem.MaxBrakeShoeForceN != 0 ? brakeSystem.MaxBrakeShoeForceN : brakeSystem.InitialMaxBrakeForceN;
+                    LoadEmptyRelayValveRatio = (brakeSystem as AirSinglePipe)?.RelayValveRatio ?? 0;
+                    LoadEmptyInshotPSI = (brakeSystem as AirSinglePipe)?.RelayValveInshotPSI ?? 0;
+                    FrictionBrakeBlendingMaxForceN = brakeSystem.InitialMaxBrakeForceN; // set the value of braking when blended with dynamic brakes
+
+                    // 3) Finally try to find the next load stage for being able to interpolate if necessary
+                    var next = float.MaxValue;
+                    foreach (var key in BrakeSystems.Keys)
+                        if (key.BrakeMode == mode && massKg <= key.MinMass)
+                            next = Math.Min(next, key.MinMass);
+                    if (next == float.MaxValue)
+                        next = max;
+
+                    if (!BrakeSystems.TryGetValue((mode, next), out var brakeSystemNext))
+                        brakeSystemNext = BrakeSystem;
+
+                    LoadFullMaxHandbrakeForceN = brakeSystemNext.InitialMaxHandbrakeForceN;
+                    LoadFullMaxBrakeForceN = BrakeShoeType != BrakeShoeTypes.Unknown && brakeSystemNext.MaxBrakeShoeForceN != 0 ? brakeSystemNext.MaxBrakeShoeForceN : brakeSystemNext.InitialMaxBrakeForceN;
+                    LoadFullRelayValveRatio = (brakeSystemNext as AirSinglePipe)?.RelayValveRatio ?? 0;
+                    LoadFullInshotPSI = (brakeSystemNext as AirSinglePipe)?.RelayValveInshotPSI ?? 0;
+                }
+            }
+            else
+            {
+                // Set up the traditional brake config style without any brake modes and freight animations
+                LoadEmptyMaxHandbrakeForceN = LoadFullMaxHandbrakeForceN = BrakeSystem.InitialMaxHandbrakeForceN;
```

Comment:
@SteelFill Yeah, seems I didn't realize that while I set the default values for the brake related freightanim variables here, I need to do the same for the Davis related ones and/or move the UpdateDavisLoadCompensation(TempMassDiffRatio) call at the end of the LoadFromWagFile into the freightanim block above it.